packages/phpstan-rules/src/Rules/PreventDuplicateClassMethodRule.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Rules;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\PrettyPrinter\Standard;
+use PHPStan\Analyser\Scope;
+use Symplify\PHPStanRules\Naming\SimpleNameResolver;
+use Symplify\PHPStanRules\ValueObject\MethodName;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\PHPStanRules\Tests\Rules\PreventDuplicateClassMethodRule\PreventDuplicateClassMethodRuleTest
+ */
+final class PreventDuplicateClassMethodRule extends AbstractSymplifyRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = 'Content of method "%s" is duplicated with method in "%s" class, use unique content instead';
+
+    /**
+     * @var string[]
+     */
+    private const PHPSTAN_GET_NODE_TYPE_METHODS = ['getNodeType', 'getNodeTypes'];
+
+    /**
+     * @var SimpleNameResolver
+     */
+    private $simpleNameResolver;
+
+    /**
+     * @var Standard
+     */
+    private $printerStandard;
+
+    /**
+     * @var array<string, string>
+     */
+    private $firstClassByName = [];
+
+    /**
+     * @var array<string, string>
+     */
+    private $contentMethodByName = [];
+
+    public function __construct(SimpleNameResolver $simpleNameResolver, Standard $printerStandard)
+    {
+        $this->simpleNameResolver = $simpleNameResolver;
+        $this->printerStandard = $printerStandard;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     * @return string[]
+     */
+    public function process(Node $node, Scope $scope): array
+    {
+        /** @var Class_|null */
+        $class = self::resolveCurrentClassWithNamespacedNameCheck($node);
+
+        if ($class === null) {
+            return [];
+        }
+
+        $className = (string) $class->namespacedName;
+        if ($this->isConstructorOrInTestClass($node, $className)) {
+            return [];
+        }
+
+        $classMethodName = (string) $node->name;
+        if (in_array($classMethodName, self::PHPSTAN_GET_NODE_TYPE_METHODS, true)) {
+            return [];
+        }
+
+        if (! $node->isPublic()) {
+            return [];
+        }
```

Comment:
Also private/protected methods should be included.

What was the reason to exclude those?

Reply:
For example, there are duplicated methods in some PHPStan rules like `$this->areNodesEqual`, can be upped to `AbstractSymplifyRule` class but that may need a construct injection with StandardPrinter

Reply:
I see. That would be problem, becuase PHPStan does not support basic Nette DI features like injects (`@requrie` in Symfony). 

These should be refactored to a new `NodeComparator` service, that would be used instead of these duplicated methods.

Reply:
I've updated to allow non-public method check, it will need more exclude in phpstan for existing codebase.